extensions/atlasmap/deployment/src/main/java/org/apache/camel/quarkus/component/atlasmap/deployment/AtlasmapProcessor.java

```diff
@@ -361,6 +75,35 @@ NativeImageResourceBuildItem resource() {
         return new NativeImageResourceBuildItem("META-INF/services/atlas/module/atlas.module");
     }
 
+    @BuildStep
+    void addDependencies(BuildProducer<IndexDependencyBuildItem> indexDependency) {
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-core"));
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-model"));
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-xml-model"));
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-java-model"));
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-json-model"));
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-csv-model"));
+        indexDependency.produce(new IndexDependencyBuildItem("io.atlasmap", "atlas-dfdl-model"));
+    }
+
+    @BuildStep
+    ReflectiveClassBuildItem registerForReflection(CombinedIndexBuildItem combinedIndex) {
+        IndexView index = combinedIndex.getIndex();
+        // register Atlasmap model classes for serialize/deserialize
+        String[] dtos = index.getKnownClasses().stream().map(ci -> ci.name().toString())
+                .filter(n -> n.startsWith("io.atlasmap.v2")
+                        || n.startsWith("io.atlasmap.actions")
+                        || n.startsWith("io.atlasmap.converters")
+                        || n.startsWith("io.atlasmap.json.v2")
+                        || n.startsWith("io.atlasmap.java.v2")
+                        || n.startsWith("io.atlasmap.xml.v2")
+                        || n.startsWith("io.atlasmap.csv.v2")
+                        || n.startsWith("io.atlasmap.dfdl.v2"))
+                .toArray(String[]::new);
+
+        return new ReflectiveClassBuildItem(true, true, dtos);
```

Comment:
An obligatory `true, true` comment from my side :) 

`new ReflectiveClassBuildItem(true, true` should be avoided as far as possible, because it bloats the amount of code the native compiler has to process thus substantially prolonging the native compilation and increasing the amount of RAM needed. 

It is very rare that a lib/framework uses both method and field reflection. Hence either `ReflectiveClassBuildItem(true, false` or `ReflectiveClassBuildItem(false, true` should be typically enough.

In this particular case, some of the classes were registered with `ReflectiveClassBuildItem(true, false` before this change and unless there is a known problem with that, we should prefer registering them like that.

So can we perhaps reduce number of `true, true` registrations in some way?

Reply:
Yeah (true, true) was the easiest way.
I'm working on this one to find the best way to reduce this.